python/ctsm/site_and_regional/neon_site.py

```diff
@@ -27,126 +27,33 @@
-        all the other cases are cloned from this case
-
-        Args:
-        self:
-            The NeonSite object
-        base_root (str):
-            root of the base_case CIME
-        res (str):
-            base_case resolution or gridname
-        compset (str):
-            base case compset
-        overwrite (bool) :
-            Flag to overwrite the case if exists
-        """
-        print("---- building a base case -------")
-        # pylint: disable=attribute-defined-outside-init
-        self.base_case_root = output_root
-        # pylint: enable=attribute-defined-outside-init
-        user_mods_dirs = [os.path.join(cesmroot, "cime_config", "usermods_dirs", "NEON", self.name)]
-        if not output_root:
-            output_root = os.getcwd()
-        case_path = os.path.join(output_root, self.name)
-
-        logger.info("base_case_name : %s", self.name)
-        logger.info("user_mods_dir  : %s", user_mods_dirs[0])
-
-        if overwrite and os.path.isdir(case_path):
-            print("Removing the existing case at: {}".format(case_path))
-            shutil.rmtree(case_path)
-
-        with Case(case_path, read_only=False) as case:
-            if not os.path.isdir(case_path):
-                print("---- creating a base case -------")
-
-                case.create(
-                    case_path,
-                    cesmroot,
-                    compset,
-                    res,
-                    run_unsupported=True,
-                    answer="r",
-                    output_root=output_root,
-                    user_mods_dirs=user_mods_dirs,
-                    driver="nuopc",
-                )
-
-                print("---- base case created ------")
-
-                # --change any config for base_case:
-                # case.set_value("RUN_TYPE","startup")
-                print("---- base case setup ------")
-                case.case_setup()
-            else:
-                # For existing case check that the compset name is correct
-                existingcompname = case.get_value("COMPSET")
-                match = re.search("^HIST", existingcompname, flags=re.IGNORECASE)
-                if re.search("^HIST", compset, flags=re.IGNORECASE) is None:
-                    expect(
-                        match is None,
-                        """Existing base case is a historical type and should not be
-                        --rerun with the --overwrite option""",
-                    )
-                else:
-                    expect(
-                        match is not None,
-                        """Existing base case should be a historical type and is not
-                        --rerun with the --overwrite option""",
-                    )
-                # reset the case
-                case.case_setup(reset=True)
-            case_path = case.get_value("CASEROOT")
-
-            if setup_only:
-                return case_path
+        if user_mods_dirs is None:
```

Comment:
Ahh this part is specific to NEON because of the path, there will be similar code to this for PLUMBER2, just with a different path.

A possible refactoring would be to have this as a generic function in the TowerSite class and you then give the specific name for. There isn't a lot of code here for that though...

Reply:
But, we should also add an error check that the directory is valid and that would be a good reason to have a generic function in TowerSite for this.

Reply:
#2412 and #2413